apriltag:
  ros__parameters:
    image_transport: raw
    family: 16h5
    size: 0.0105
    tag:
      ids:
      - 0
      - 1
      - 2
      - 3
      - 4
      - 5
      - 6
      - 7
      - 8
      - 10
      frames:
      - tag16h5:0
      - tag16h5:1
      - tag16h5:2
      - tag16h5:3
      - tag16h5:4
      - tag16h5:5
      - tag16h5:6
      - tag16h5:7
      - tag16h5:8
      - tag16h5:10
      sizes:
      - 0.0105
      - 0.0105
      - 0.0105
      - 0.0105
      - 0.0105
      - 0.0105
      - 0.0105
      - 0.0105
      - 0.0105
      - 0.0105
    bundle:
      name: dls_bundle_centroid
      ids:
      - 0
      - 1
      - 2
      - 3
      - 4
      - 5
      - 6
      - 7
      # Bundle center offsets FROM each tag TO the center
      # 3x3 grid of 10.5mm tags with 7.5mm spacing (18mm between centers)
      # Bundle center is at the center of the grid
      # All tags have same orientation, different positions
      poses:
        0:  # Top-left tag - 18mm right, 18mm up to bundle center
          translation:
          - 0.018   # +18mm to center
          - -0.018  # -18mm to center
          - 0.0
          rotation_xyzw:
          - 0.0
          - 0.0
          - 0.0
          - 1.0
        1:  # Top-center tag - 0mm left/right, 18mm up to bundle center
          translation:
          - 0.0     # 0mm to center
          - -0.018  # -18mm to center
          - 0.0
          rotation_xyzw:
          - 0.0
          - 0.0
          - 0.0
          - 1.0
        2:  # Top-right tag - 18mm left, 18mm up to bundle center
          translation:
          - -0.018  # -18mm to center
          - -0.018  # -18mm to center
          - 0.0
          rotation_xyzw:
          - 0.0
          - 0.0
          - 0.0
          - 1.0
        3:  # Middle-right tag - 18mm left, 0mm up/down to bundle center
          translation:
          - -0.018  # -18mm to center
          - 0.0     # 0mm to center
          - 0.0
          rotation_xyzw:
          - 0.0
          - 0.0
          - 0.0
          - 1.0
        4:  # Bottom-right tag - 18mm left, 18mm down to bundle center
          translation:
          - -0.018  # -18mm to center
          - 0.018   # +18mm to center
          - 0.0
          rotation_xyzw:
          - 0.0
          - 0.0
          - 0.0
          - 1.0
        5:  # Bottom-center tag - 0mm left/right, 18mm down to bundle center
          translation:
          - 0.0     # 0mm to center
          - 0.018   # +18mm to center
          - 0.0
          rotation_xyzw:
          - 0.0
          - 0.0
          - 0.0
          - 1.0
        6:  # Bottom-left tag - 18mm right, 18mm down to bundle center
          translation:
          - 0.018   # +18mm to center
          - 0.018   # +18mm to center
          - 0.0
          rotation_xyzw:
          - 0.0
          - 0.0
          - 0.0
          - 1.0
        7:  # Middle-left tag - 18mm right, 0mm up/down to bundle center
          translation:
          - 0.018   # +18mm to center
          - 0.0     # 0mm to center
          - 0.0
          rotation_xyzw:
          - 0.0
          - 0.0
          - 0.0
          - 1.0
